tests.nix

```diff
@@ -0,0 +1,58 @@
+let
+  sources = import nix/sources.nix;
+in
+{ pkgs ? import sources.release2015 {}
+, pypiData ? sources.pypi-deps-db
+, mach-nix ? import sources.mach-nix { inherit pkgs pypiData; }
+, tahoe-lafs-source ? "tahoe-lafs"
+, tahoe-lafs-repo ? sources.${tahoe-lafs-source}
+, privatestorage ? import ./. {
+    inherit pkgs pypiData mach-nix;
+    inherit tahoe-lafs-repo;
+  }
+, hypothesisProfile ? null
+, collectCoverage ? false
+, testSuite ? null
+, trialArgs ? null
+,
+}:
+  let
+    inherit (pkgs) lib;
+    inherit (privatestorage) zkapauthorizer;
+    hypothesisProfile' = if hypothesisProfile == null then "default" else hypothesisProfile;
+    defaultTrialArgs = [ "--rterrors" ] ++ (lib.optional (! collectCoverage) "--jobs=$NIX_BUILD_CORES");
+    trialArgs' = if trialArgs == null then defaultTrialArgs else trialArgs;
+    extraTrialArgs = builtins.concatStringsSep " " trialArgs';
+    testSuite' = if testSuite == null then "_zkapauthorizer" else testSuite;
+
+    python = mach-nix.mkPython {
+      inherit (zkapauthorizer.meta.mach-nix) python providers;
+      requirements =
+        builtins.readFile ./requirements/test.in;
+      packagesExtra = [ zkapauthorizer ];
+      _.hypothesis.postUnpack = "";
+    };
+  in
+    pkgs.runCommand "zkapauthorizer-tests" {
+      passthru = {
+        inherit python;
+      };
```

Comment:
This is exposing the python environment with test dependencies, for use in `shell.nix`.